plugins/modules/cloud/xenserver/xenserver_guest.py

```diff
@@ -1807,7 +1873,7 @@ def main():
         cdrom=dict(
             type='dict',
             options=dict(
-                type=dict(type='str', choices=['none', 'iso']),
+                type=dict(type='str', choices=['none', 'iso']),  # @TODO: required ?
```

Comment:
Not required. `iso_name` can be set without `type` in which case `type` is implicitly set to `iso`. If nothing is set, `type` is assumed to be `none`. This seems to make `type` useless but `type` is kept as is for now because there is an unimplemented type `host`. Once support for `host` type is implemented, `type` will have to be explicitly set.

Reply:
Shouldn't we have a `default="iso"` until then?

Reply:
No. If `type` has value `iso`, than `iso_name` must be specified. There is no way to tell Ansible "I want an empty CDROM", or better said, you would always have to specify `type: none`.

Reply:
Anyway, that should not be in this PR. :-)